src/BenchmarkDotNet/Filters/GlobFilter.cs

```diff
@@ -11,16 +13,19 @@ public class GlobFilter : IFilter
     {
         private readonly Regex[] patterns;
 
-        public GlobFilter(string[] patterns) => this.patterns = ToRegex(patterns);
+        public GlobFilter(string[] patterns) => this.patterns = ToRegex(AddWildcardCharacterToEnd(patterns));
 
         public bool Predicate(BenchmarkCase benchmarkCase)
         {
             var benchmark = benchmarkCase.Descriptor.WorkloadMethod;
-            string fullBenchmarkName = benchmarkCase.Descriptor.GetFilterName();
+            string fullBenchmarkName = FullNameProvider.GetBenchmarkName(benchmarkCase);
 
             return patterns.Any(pattern => pattern.IsMatch(fullBenchmarkName));
         }
 
+        internal static string[] AddWildcardCharacterToEnd(string[] patterns)
+            => Array.ConvertAll(patterns, pattern => pattern.EndsWith("*") ? pattern : $"{pattern}*");
```

Comment:
This might have some unintended side effects. Example: let's say that the user provides `--filter $namespace` without asterisk at the end. This filter is wrong, and it should return 0 benchmarks, but by adding `*` here we make it "correct".

To remain the backward compatibility, we could do something like this:

```cs
string nameWithoutArgs = benchmarkCase.Descriptor.GetFilterName();
string fullBenchmarkName = FullNameProvider.GetBenchmarkName(benchmarkCase)

return patterns.Any(pattern => pattern.IsMatch(fullBenchmarkName) || pattern.IsMatch(nameWithoutArgs));
```

@blouflashdb what do you think?

Reply:
Ah yes, I was not thinking about that sideeffect. I like your idea to remain backward compatibility. Going to change it. Thanks for the review.